Bottom: 26cd219a844320c53cd246827e8ee8f18ec70831
Top:    27b0315ec3770841902a29a954627da1fa0862af
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-09 16:49:16 +0800

Refresh of formal-patch

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 8eb9afb..5b6a546 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1886,7 +1886,7 @@ static int tun_set_iff(struct net *net, struct file *file, struct ifreq *ifr)
 
 		tun = netdev_priv(dev);
 		tun->dev = dev;
-		tun->flags = flags;
+		tun->flags = flags | IFF_TX_RING;
 		tun->txflt.count = 0;
 		tun->vnet_hdr_sz = sizeof(struct virtio_net_hdr);
 
diff --git a/include/uapi/linux/if_tun.h b/include/uapi/linux/if_tun.h
index 39061f1..029cdda 100644
--- a/include/uapi/linux/if_tun.h
+++ b/include/uapi/linux/if_tun.h
@@ -61,7 +61,7 @@
 #define IFF_TUN		0x0001
 #define IFF_TAP		0x0002
 #define IFF_NO_PI	0x1000
-#define IFF_TX_RING	0x0004
+#define IFF_TX_RING	0x4000
 /* This flag has no real effect */
 #define IFF_ONE_QUEUE	0x2000
 #define IFF_VNET_HDR	0x4000
